src/webgpu/api/validation/capability_checks/limits/maxStorageBufferBindingSize.spec.ts

```diff
@@ -0,0 +1,166 @@
+  const align = device.limits.minUniformBufferOffsetAlignment;
+  switch (bufferPart) {
+    case 'wholeBuffer':
+      return { size, offset: 0 };
+    case 'biggerBufferWithOffset':
+      return { size: size + align, offset: align };
+  }
+}
+
+const kStorageBufferRequiredSizeAlignment = 4;
+
+function getDeviceLimitToRequest(
+  limitValueTest: LimitValueTest,
+  defaultLimit: number,
+  maximumLimit: number
+) {
+  switch (limitValueTest) {
+    case 'atDefault':
+      return defaultLimit;
+    case 'underDefault':
+      return defaultLimit - kStorageBufferRequiredSizeAlignment;
+    case 'betweenDefaultAndMaximum':
+      return Math.floor((defaultLimit + maximumLimit) / 2);
+    case 'atMaximum':
+      return maximumLimit;
+    case 'overMaximum':
+      return maximumLimit + kStorageBufferRequiredSizeAlignment;
+  }
+}
+
+function getTestValue(testValueName: TestValue, requestedLimit: number) {
+  switch (testValueName) {
+    case 'atLimit':
+      return roundDown(requestedLimit, kStorageBufferRequiredSizeAlignment);
+    case 'overLimit':
+      // Note: the requestedLimit might not meet alignment requirements.
+      return roundDown(
+        requestedLimit + kStorageBufferRequiredSizeAlignment,
+        kStorageBufferRequiredSizeAlignment
+      );
+  }
+}
+
+function getDeviceLimitToRequestAndValueToTest(
+  limitValueTest: LimitValueTest,
+  testValueName: TestValue,
+  defaultLimit: number,
+  maximumLimit: number
+) {
+  const requestedLimit = getDeviceLimitToRequest(limitValueTest, defaultLimit, maximumLimit);
+  return {
+    requestedLimit,
+    testValue: getTestValue(testValueName, requestedLimit),
+  };
+}
+
+const limit = 'maxStorageBufferBindingSize';
+export const { g, description } = makeLimitTestGroup(limit);
+
+g.test('createBindGroup,at_over')
+  .desc(`Test using createBindGroup at and over ${limit} limit`)
+  .params(kLimitBaseParams.combine('bufferPart', kBufferPartsKeys))
+  .fn(async t => {
+    const { limitTest, testValueName, bufferPart } = t.params;
+    const { adapter, defaultLimit, maximumLimit } = await t.getAdapterAndLimits();
+    const { requestedLimit, testValue } = getDeviceLimitToRequestAndValueToTest(
+      limitTest,
+      testValueName,
+      defaultLimit,
+      maximumLimit
+    );
+
+    await t.testDeviceWithSpecificLimits(
+      adapter,
+      requestedLimit,
+      testValue,
+      async ({ device, testValue, shouldError }) => {
+        const bindGroupLayout = device.createBindGroupLayout({
+          entries: [
+            {
+              binding: 0,
+              visibility: GPUShaderStage.COMPUTE,
+              buffer: { type: 'storage' },
+            },
+          ],
+        });
+
+        const { size, offset } = getSizeAndOffsetForBufferPart(device, bufferPart, testValue);
+        device.pushErrorScope('out-of-memory');
+        const storageBuffer = device.createBuffer({
```

Comment:
Can probably wrap with `trackForCleanup` and then we don't need the `destroy` call at the end.

Reply:
done